test/credentials_test.dart

```diff
@@ -38,6 +39,8 @@ Future<void> main([List<String>? args]) async {
 
     expect(user1, user2);
     expect(user1, isNot(user3));
+    expect(user1.provider, AuthProviderType.anonymous);
+    expect(user3.provider, AuthProviderType.anonymous);
```

Comment:
After login with with anonymous `reuseCredentials: false`  we receive `AuthProviderType.anonymous` instead of `AuthProviderType.anonymousNoReuse`. It is because for the server both credentials are "anon-user" and this is the way they are mapped in `provider_type_from_enum`. @nirinchev do you think we have to create an issue about that?

Reply:
No, apparently that's intentional. We should probably remove the noreuse provider type. As that's not something users should ever deal with.

Reply:
Maybe we have to think about it, because when we pass the credentials to core they expect `anonymousNoReuse` to know whether to reuse the users or not.

Reply:
Right, but we construct the credentials by calling `realmCore.createAppCredentialsAnonymous` - we don't provide the actual enum value to anyone. We store it as a field on the dart credentials, but I don't believe that's actually ever used anywhere and we might as well turn that into a property that reads the value from Core.

Reply:
OK, I made anonymousNoReuse hidden.